src/bpmncwpverify/bpmn.py

```diff
@@ -89,29 +150,75 @@ def __init__(
         super().__init__(element)
         self.source_node: Node
         self.target_node: Node
+        self.is_leaf: bool = False
+
+    @abstractmethod
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        pass
 
 
 class SequenceFlow(Flow):
     def __init__(self, element: Element):
         super().__init__(element)
 
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        visitor.visitSequenceFlow(self)
+        if not self.is_leaf:
+            self.target_node.accept(visitor)
+        visitor.endVisitSequenceFlow(self)
+
 
 class MessageFlow(Flow):
     def __init__(self, element: Element):
         super().__init__(element)
 
+    def accept(self, visitor: "BpmnVisitor") -> None:
+        visitor.visitMessageFlow(self)
+        if not self.is_leaf:
+            self.target_node.accept(visitor)
+        visitor.endVisitMessageFlow(self)
 
+
+###################
 # Process class
+###################
 class Process(BpmnElement):
     def __init__(self, element: Element):
         super().__init__(element)
-        self.elements: Dict[str, Node] = {}
         self.flows: Dict[str, Flow] = {}
+        self._elements: Dict[str, Node] = {}
+        self._start_states: Dict[str, StartEvent] = {}
+
+    def __setitem__(self, key: str, node: Node) -> None:
```

Comment:
I separated start states and all other elements into two different attributes so that I could easily loop over start states (in order to know where to begin traversals, etc.). Then, so that I don't have to have a conditional everywhere that determines if something belongs in _elements or _start_states, I moved this into the __setitem__ and __getitem__ functions.